Incorrect types for native errors subtypes

**TypeScript Version:**  master (3.4.0)

<!-- Search terms you tried before logging this (so others can find this issue more easily) -->
**Search Terms:** ErrorConstructor

**Code**
```ts
// In Node, ErrorConstructor is augmented with extra properties. Excerpted below.
interface ErrorConstructor {
  captureStackTrace(targetObject: Object, constructorOpt?: Function): void;
}

function myFunction(x: ErrorConstructor) { return true; }
myFunction(Error); // OK

// Error: Property 'captureStackTrace' is missing in type 'RangeErrorConstructor' 
// but required in type 'ErrorConstructor'.
myFunction(RangeError); 
```

**Actual behavior:**
Errors, as indicated in the comment.

**Expected behavior:**

I would expect calling `myFunction(RangeError)` to work. 

[According to the spec](https://tc39.github.io/ecma262/#sec-properties-of-the-nativeerror-constructors), the constructor functions for the native errors (`RangeError`, `SyntaxError`, etc) have the base `Error` constructor as their [[Prototype]]. Therefore, I would expect that any augmentations to the base `ErrorConstructor` type would be visible on the interfaces for the other native error constructors. My first instinct was that this would be accomplished by setting up those child interfaces in `lib.es5.d.ts` to extend `ErrorConstructor`, i.e.:

```ts
// Proposed lib.es5.d.ts
interface RangeErrorConstructor extends ErrorConstructor { /* ... */ } 
interface SyntaxErrorConstructor extends ErrorConstructor { /* ... */ } 
// etc.
```

Today, the other error constructor interfaces don't extend `ErrorConstructor`, so the type system doesn't know that `RangeErrorConstructor` also has a `captureStackTrace` function.

However, when I tried to edit `lib.es5.d.ts` to set up the `extends` clause, I ran into this error, from the Node typings:

```
error TS2510: Base constructors must all have the same return type.

1861     export class Recoverable extends SyntaxError {
                                          ~~~~~~~~~~~
```

I'm not sure what the right way to fix this is, but I do think the behavior in my first code snippet is a bug, and I would hope there's some way to address it (that also still allows the various native error constructors to themselves be subclassable).

**Playground Link:** 
[Here](http://www.typescriptlang.org/play/index.html#src=%2F%2F%20In%20Node%2C%20ErrorConstructor%20is%20augmented%20with%20extra%20properties.%20Excerpted%20below.%0D%0Ainterface%20ErrorConstructor%20%7B%0D%0A%20%20captureStackTrace(targetObject%3A%20Object%2C%20constructorOpt%3F%3A%20Function)%3A%20void%3B%0D%0A%7D%0D%0A%0D%0Afunction%20myFunction(x%3A%20ErrorConstructor)%20%7B%20return%20true%3B%20%7D%0D%0AmyFunction(Error)%3B%20%2F%2F%20OK%0D%0AmyFunction(RangeError)%3B%20%2F%2F%20Error%0D%0A%0D%0Ainterface%20Error%20%7B%0D%0A%20%20%20%20name%3A%20string%3B%0D%0A%20%20%20%20message%3A%20string%3B%0D%0A%20%20%20%20stack%3F%3A%20string%3B%0D%0A%7D%0D%0A%0D%0A%2F%2F%20This%20is%20my%20proposed%20fix%2C%20but%20see%20the%20error%20it%20causes%20with%20Recoverable.%0D%0Ainterface%20SyntaxErrorConstructor%20extends%20ErrorConstructor%20%20%7B%0D%0A%20%20%20%20new(message%3F%3A%20string)%3A%20SyntaxError%3B%0D%0A%20%20%20%20(message%3F%3A%20string)%3A%20SyntaxError%3B%0D%0A%20%20%20%20readonly%20prototype%3A%20SyntaxError%3B%0D%0A%7D%0D%0A%0D%0Aclass%20Recoverable%20extends%20SyntaxError%20%7B%0D%0A%0D%0A%7D)

--- a/src/lib/es5.d.ts
+++ b/src/lib/es5.d.ts
@@ -966,7 +966,7 @@ declare var Error: ErrorConstructor;
 interface EvalError extends Error {
 }
 
-interface EvalErrorConstructor {
+interface EvalErrorConstructor extends ErrorConstructor {
     new(message?: string): EvalError;
     (message?: string): EvalError;
     readonly prototype: EvalError;
@@ -977,7 +977,7 @@ declare var EvalError: EvalErrorConstructor;
 interface RangeError extends Error {
 }
 
-interface RangeErrorConstructor {
+interface RangeErrorConstructor extends ErrorConstructor {
     new(message?: string): RangeError;
     (message?: string): RangeError;
     readonly prototype: RangeError;
@@ -988,7 +988,7 @@ declare var RangeError: RangeErrorConstructor;
 interface ReferenceError extends Error {
 }
 
-interface ReferenceErrorConstructor {
+interface ReferenceErrorConstructor extends ErrorConstructor {
     new(message?: string): ReferenceError;
     (message?: string): ReferenceError;
     readonly prototype: ReferenceError;
@@ -999,7 +999,7 @@ declare var ReferenceError: ReferenceErrorConstructor;
 interface SyntaxError extends Error {
 }
 
-interface SyntaxErrorConstructor {
+interface SyntaxErrorConstructor extends ErrorConstructor {
     new(message?: string): SyntaxError;
     (message?: string): SyntaxError;
     readonly prototype: SyntaxError;
@@ -1010,7 +1010,7 @@ declare var SyntaxError: SyntaxErrorConstructor;
 interface TypeError extends Error {
 }
 
-interface TypeErrorConstructor {
+interface TypeErrorConstructor extends ErrorConstructor {
     new(message?: string): TypeError;
     (message?: string): TypeError;
     readonly prototype: TypeError;
@@ -1021,7 +1021,7 @@ declare var TypeError: TypeErrorConstructor;
 interface URIError extends Error {
 }
 
-interface URIErrorConstructor {
+interface URIErrorConstructor extends ErrorConstructor {
     new(message?: string): URIError;
     (message?: string): URIError;
     readonly prototype: URIError;

--- a/tests/baselines/reference/errorConstructorSubtypes.js
+++ b/tests/baselines/reference/errorConstructorSubtypes.js
@@ -1,0 +1,18 @@
+//// [errorConstructorSubtypes.ts]
+// In Node, ErrorConstructor is augmented with extra properties. Excerpted below.
+interface ErrorConstructor {
+  captureStackTrace(targetObject: Object, constructorOpt?: Function): void;
+}
+
+declare var x: ErrorConstructor
+x = Error; // OK
+x = RangeError;
+new x().message
+x.captureStackTrace
+
+
+//// [errorConstructorSubtypes.js]
+x = Error; // OK
+x = RangeError;
+new x().message;
+x.captureStackTrace;

--- a/tests/baselines/reference/errorConstructorSubtypes.symbols
+++ b/tests/baselines/reference/errorConstructorSubtypes.symbols
@@ -1,0 +1,35 @@
+=== tests/cases/compiler/errorConstructorSubtypes.ts ===
+// In Node, ErrorConstructor is augmented with extra properties. Excerpted below.
+interface ErrorConstructor {
+>ErrorConstructor : Symbol(ErrorConstructor, Decl(lib.es5.d.ts, --, --), Decl(errorConstructorSubtypes.ts, 0, 0))
+
+  captureStackTrace(targetObject: Object, constructorOpt?: Function): void;
+>captureStackTrace : Symbol(ErrorConstructor.captureStackTrace, Decl(errorConstructorSubtypes.ts, 1, 28))
+>targetObject : Symbol(targetObject, Decl(errorConstructorSubtypes.ts, 2, 20))
+>Object : Symbol(Object, Decl(lib.es5.d.ts, --, --), Decl(lib.es5.d.ts, --, --))
+>constructorOpt : Symbol(constructorOpt, Decl(errorConstructorSubtypes.ts, 2, 41))
+>Function : Symbol(Function, Decl(lib.es5.d.ts, --, --), Decl(lib.es5.d.ts, --, --))
+}
+
+declare var x: ErrorConstructor
+>x : Symbol(x, Decl(errorConstructorSubtypes.ts, 5, 11))
+>ErrorConstructor : Symbol(ErrorConstructor, Decl(lib.es5.d.ts, --, --), Decl(errorConstructorSubtypes.ts, 0, 0))
+
+x = Error; // OK
+>x : Symbol(x, Decl(errorConstructorSubtypes.ts, 5, 11))
+>Error : Symbol(Error, Decl(lib.es5.d.ts, --, --), Decl(lib.es5.d.ts, --, --))
+
+x = RangeError;
+>x : Symbol(x, Decl(errorConstructorSubtypes.ts, 5, 11))
+>RangeError : Symbol(RangeError, Decl(lib.es5.d.ts, --, --), Decl(lib.es5.d.ts, --, --))
+
+new x().message
+>new x().message : Symbol(Error.message, Decl(lib.es5.d.ts, --, --))
+>x : Symbol(x, Decl(errorConstructorSubtypes.ts, 5, 11))
+>message : Symbol(Error.message, Decl(lib.es5.d.ts, --, --))
+
+x.captureStackTrace
+>x.captureStackTrace : Symbol(ErrorConstructor.captureStackTrace, Decl(errorConstructorSubtypes.ts, 1, 28))
+>x : Symbol(x, Decl(errorConstructorSubtypes.ts, 5, 11))
+>captureStackTrace : Symbol(ErrorConstructor.captureStackTrace, Decl(errorConstructorSubtypes.ts, 1, 28))
+

--- a/tests/baselines/reference/errorConstructorSubtypes.types
+++ b/tests/baselines/reference/errorConstructorSubtypes.types
@@ -1,0 +1,33 @@
+=== tests/cases/compiler/errorConstructorSubtypes.ts ===
+// In Node, ErrorConstructor is augmented with extra properties. Excerpted below.
+interface ErrorConstructor {
+  captureStackTrace(targetObject: Object, constructorOpt?: Function): void;
+>captureStackTrace : (targetObject: Object, constructorOpt?: Function) => void
+>targetObject : Object
+>constructorOpt : Function
+}
+
+declare var x: ErrorConstructor
+>x : ErrorConstructor
+
+x = Error; // OK
+>x = Error : ErrorConstructor
+>x : ErrorConstructor
+>Error : ErrorConstructor
+
+x = RangeError;
+>x = RangeError : RangeErrorConstructor
+>x : ErrorConstructor
+>RangeError : RangeErrorConstructor
+
+new x().message
+>new x().message : string
+>new x() : Error
+>x : ErrorConstructor
+>message : string
+
+x.captureStackTrace
+>x.captureStackTrace : (targetObject: Object, constructorOpt?: Function) => void
+>x : ErrorConstructor
+>captureStackTrace : (targetObject: Object, constructorOpt?: Function) => void
+

--- a/tests/cases/compiler/errorConstructorSubtypes.ts
+++ b/tests/cases/compiler/errorConstructorSubtypes.ts
@@ -1,0 +1,11 @@
+// @lib: es5,dom
+// In Node, ErrorConstructor is augmented with extra properties. Excerpted below.
+interface ErrorConstructor {
+  captureStackTrace(targetObject: Object, constructorOpt?: Function): void;
+}
+
+declare var x: ErrorConstructor
+x = Error; // OK
+x = RangeError;
+new x().message
+x.captureStackTrace

